ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/internalresponse/InternalResponseHandler.java

```diff
@@ -14,40 +14,42 @@
 
 import tech.pegasys.ethsigner.http.HttpResponseFactory;
 import tech.pegasys.ethsigner.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.jsonrpc.exception.JsonRpcException;
 import tech.pegasys.ethsigner.jsonrpc.response.JsonRpcSuccessResponse;
 import tech.pegasys.ethsigner.requesthandler.BodyProvider;
 import tech.pegasys.ethsigner.requesthandler.JsonRpcBody;
 import tech.pegasys.ethsigner.requesthandler.JsonRpcRequestHandler;
 
 import io.netty.handler.codec.http.HttpResponseStatus;
-import io.vertx.core.http.HttpServerRequest;
 import io.vertx.core.json.Json;
+import io.vertx.ext.web.RoutingContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class InternalResponseHandler extends JsonRpcRequestHandler {
+public class InternalResponseHandler implements JsonRpcRequestHandler {
 
   private static final Logger LOG = LoggerFactory.getLogger(InternalResponseHandler.class);
 
+  private final HttpResponseFactory responder;
   private final BodyProvider responseBodyProvider;
 
   public InternalResponseHandler(
       final HttpResponseFactory responder, final BodyProvider responseBodyProvider) {
-    super(responder);
+    this.responder = responder;
     this.responseBodyProvider = responseBodyProvider;
   }
 
   @Override
-  public void handle(final HttpServerRequest httpServerRequest, final JsonRpcRequest rpcRequest) {
+  public void handle(final RoutingContext context, final JsonRpcRequest rpcRequest) {
```

Comment:
Using the context.fail is really nice (and how Vertx kinda expects to operate) - but do we want to expose the context the whole way down?

Reply:
We don't need the entire context, so could just pass in the httpServerRequest as before and the fail method as method reference. Alternatively could just throw an exception instead. However the failure way is more the vertx way and fits with how the timeout handler works so if we did something different would need to still handle both ways in the error handler. Only small change though.